teamcity: failed tests on master: testrace/TestStoreRangeUpReplicate

The following tests appear to have failed:

[#43766](https://teamcity.cockroachdb.com/viewLog.html?buildId=43766):

```
--- FAIL: testrace/TestStoreRangeUpReplicate (0.000s)
	client_raft_test.go:969: expected 0 reservations, but found 1

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/storage/client_raft_test.go
+++ b/pkg/storage/client_raft_test.go
@@ -953,6 +953,9 @@ func TestStoreRangeUpReplicate(t *testing.T) {
 			if r == nil {
 				return errors.Errorf("expected replica for keys \"a\" - \"b\"")
 			}
+			if n := s.ReservationCount(); n != 0 {
+				return errors.Errorf("expected 0 reservations, but found %d", n)
+			}
 		}
 		return nil
 	})
@@ -965,9 +968,6 @@ func TestStoreRangeUpReplicate(t *testing.T) {
 		generated += m.RangeSnapshotsGenerated.Count()
 		normalApplied += m.RangeSnapshotsNormalApplied.Count()
 		preemptiveApplied += m.RangeSnapshotsPreemptiveApplied.Count()
-		if n := s.ReservationCount(); n != 0 {
-			t.Fatalf("expected 0 reservations, but found %d", n)
-		}
 	}
 	if generated == 0 {
 		t.Fatalf("expected at least 1 snapshot, but found 0")

